common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java

```diff
@@ -66,15 +73,27 @@ public static URL appendPathToURL(@NonNull final URL urlToAppend,
         final CommonURIBuilder builder = new CommonURIBuilder(urlToAppend.toString());
         final List<String> pathSegments = builder.getPathSegments();
 
-        final List<String> combinedPathSegments = new ArrayList<>(pathSegments);
+        final List<String> combinedPathSegments = new ArrayList<>();
+
+        // TODO check this with MSAL team
+        // Add all non-empty path segments from the base URL
+        for (final String path : pathSegments) {
+            if (!StringUtil.isNullOrEmpty(path)) {
+                combinedPathSegments.add(path);
```

Comment:
This seems fine to me, although perhaps the caller should decide whether or not remove them? Not sure, maybe we can add another util method that takes the pathSegments and returns a list without the empty ones.